src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/AD/ADStoreCtx.cs

```diff
@@ -2418,12 +2421,22 @@ protected virtual void LoadDomainInfo()
             this.dnsHostName = ADUtils.GetServerName(this.ctxBase);
 
             // Pull the requested port number
-            Uri ldapUri = new Uri(this.ctxBase.Path);
-            int port = ldapUri.Port != -1 ? ldapUri.Port : (ldapUri.Scheme.ToUpperInvariant() == "LDAPS" ? 636 : 389);
+            int port = LdapDefaultPort;
+            if (Uri.TryCreate(ctxBase.Path, UriKind.Absolute, out Uri ldapUri))
+            {
+                if (ldapUri.Port != -1)
+                {
+                    port = ldapUri.Port;
+                }
+                else if (string.Equals(ldapUri.Scheme, "LDAPS", StringComparison.OrdinalIgnoreCase))
+                {
+                    port = LdapsDefaultPort;
+                }
+            }
 
             string dnsDomainName = "";
 
-            using (DirectoryEntry rootDse = new DirectoryEntry("LDAP://" + this.dnsHostName + ":" + port + "/rootDse", "", "", AuthenticationTypes.Anonymous))
+            using (DirectoryEntry rootDse = new DirectoryEntry($"LDAP://{this.dnsHostName}:{port}/rootDse", "", "", AuthenticationTypes.Anonymous))
```

Comment:
I know it wasn't a reported problem, but double checking we are OK to add the default port values here when not specified by the user?  Was that an important part of the original fix?

Reply:
Since this code only interacts with a dc it is safe to use default ports, ie., 389 for regular LDAP and 636 for secure version of it.